evaluation/app_chat_target.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+import os
+from pathlib import Path
+
+from dotenv import load_dotenv
+from pyrit.chat_message_normalizer import ChatMessageNop, ChatMessageNormalizer
+from pyrit.common import default_values, net_utility
+from pyrit.memory import MemoryInterface
+from pyrit.models import (
+    ChatMessage,
+    PromptRequestResponse,
+    construct_response_from_request,
+)
+from pyrit.prompt_target import PromptChatTarget
+
+from evaluation.utils import load_config
+
+load_dotenv()
+
+EVALUATION_DIR = Path(__file__).parent
+
+logger = logging.getLogger(__name__)
+
+
+class AppChatTarget(PromptChatTarget):
+
+    BACKEND_URI: str = os.environ.get("BACKEND_URI", "").rstrip("/") + "/ask"
+
+    def __init__(
+        self,
+        *,
+        endpoint_uri: str = None,
+        chat_message_normalizer: ChatMessageNormalizer = ChatMessageNop(),
+        memory: MemoryInterface = None,
+    ) -> None:
+
+        PromptChatTarget.__init__(self, memory=memory)
+
+        self.endpoint_uri: str = default_values.get_required_value(
+            env_var_name=self.BACKEND_URI, passed_value=endpoint_uri
+        )
+        self.chat_message_normalizer = chat_message_normalizer
+
+    async def send_prompt_async(self, *, prompt_request: PromptRequestResponse) -> PromptRequestResponse:
+
+        self._validate_request(prompt_request=prompt_request)
+        request = prompt_request.request_pieces[0]
+
+        messages = self._memory.get_chat_messages_with_conversation_id(conversation_id=request.conversation_id)
+
+        messages.append(request.to_chat_message())
+
+        logger.info(f"Sending the following prompt to the prompt target: {request}")
+
+        resp_text = await self._complete_chat_async(
+            messages=messages,
+        )
+
+        if not resp_text:
+            raise ValueError("The chat returned an empty response.")
+
+        logger.info(f'Received the following response from the prompt target "{resp_text}"')
+        return construct_response_from_request(request=request, response_text_pieces=[resp_text])
+
+    async def _complete_chat_async(
+        self,
+        messages: list[ChatMessage],
+    ) -> str:
+
+        headers = self._get_headers()
+        payload = self._construct_http_body(messages)
+
+        response = await net_utility.make_request_and_raise_if_error_async(
+            endpoint_uri=self.endpoint_uri, method="POST", request_body=payload, headers=headers
+        )
```

Comment:
Not a huge fan of using a util from an external library for something as simple as that, but ok :rofl: